Zend/zend_compile.c

```diff
@@ -8067,6 +8067,27 @@ static zend_string *zend_begin_method_decl(zend_op_array *op_array, zend_string
 		zend_error(E_COMPILE_WARNING, "Private methods cannot be final as they are never overridden by other classes");
 	}
 
+	if ((fn_flags & ZEND_ACC_ABSTRACT) &&
+		!(ce->ce_flags & (ZEND_ACC_EXPLICIT_ABSTRACT_CLASS|ZEND_ACC_TRAIT)) &&
+		!in_interface
```

Comment:
Since you already have the ce_flags check, I'd just include the `ZEND_ACC_INTERFACE` flag.

Reply:
done